<?xml version="1.0" encoding="UTF-8"?>
<class xmlns="http://xml.phpdox.net/src" full="JakubOnderka\PhpParallelLint\NotImplementCallbackException" namespace="JakubOnderka\PhpParallelLint" name="NotImplementCallbackException" abstract="false" final="false" start="79" end="93">
  <file path="//var/www/html/vendor/php-parallel-lint/php-parallel-lint/src" file="exceptions.php" realpath="/var/www/html/vendor/php-parallel-lint/php-parallel-lint/src/exceptions.php" size="1866" time="2022-02-21T12:50:22+00:00" unixtime="1645447822" sha1="25a01ed8b22afc9c385d21378fc3a15a95423415" relative="ar/www/html/vendor/php-parallel-lint/php-parallel-lint/src/exceptions.php"/>
  <extends full="JakubOnderka\PhpParallelLint\Exception" namespace="JakubOnderka\PhpParallelLint" name="Exception"/>
  <member name="className" visibility="protected" static="false" line="81"/>
  <constructor name="__construct" start="83" end="87" abstract="false" final="false" static="false" visibility="public">
    <parameter name="className" byreference="false" type="{unknown}"/>
  </constructor>
  <method name="getClassName" start="89" end="92" abstract="false" final="false" static="false" visibility="public"/>
  <parent full="JakubOnderka\PhpParallelLint\Exception" namespace="JakubOnderka\PhpParallelLint" name="Exception">
    <extends full="Exception" namespace="" name="Exception"/>
    <implements full="JsonSerializable" namespace="" name="JsonSerializable"/>
    <method name="jsonSerialize" start="9" end="16" abstract="false" final="false" static="false" visibility="public"/>
  </parent>
  <parent full="Exception" namespace="" name="Exception">
    <member name="message" static="false" visibility="protected">
    <default>null</default>
    <docblock>
      <description compact="The exception message"/>
      <var type="string"/>
    </docblock>
  </member>
    <member name="code" static="false" visibility="protected">
    <default>null</default>
    <docblock>
      <description compact="The exception code"/>
      <var type="int"/>
    </docblock>
  </member>
    <member name="file" static="false" visibility="protected">
    <default>null</default>
    <docblock>
      <description compact="The filename where the exception was created"/>
      <var type="string"/>
    </docblock>
  </member>
    <member name="line" static="false" visibility="protected">
    <default>null</default>
    <docblock>
      <description compact="The line where the exception was created"/>
      <var type="int"/>
    </docblock>
  </member>
    <constructor name="__construct" abstract="false" static="false" visibility="public" final="false">
    <docblock>
      <description compact="Construct the exception"/>
      <return type="void"/>
    </docblock>
    <parameter name="message" optional="true" byreference="false" type="string"/>
    <parameter name="code" optional="true" byreference="false" type="int"/>
    <parameter name="previous" optional="true" byreference="false" type="object" class="Throwable"/>
  </constructor>
    <method name="getMessage" abstract="false" static="false" visibility="public" final="true">
    <docblock>
      <description compact="Gets the Exception message"/>
      <return type="string"/>
    </docblock>
  </method>
    <method name="getPrevious" abstract="false" static="false" visibility="public" final="true">
    <docblock>
      <description compact="Returns previous Exception"/>
      <return type="Throwable"/>
    </docblock>
  </method>
    <method name="getCode" abstract="false" static="false" visibility="public" final="true">
    <docblock>
      <description compact="Gets the Exception code"/>
      <return type="mixed"/>
    </docblock>
  </method>
    <method name="getFile" abstract="false" static="false" visibility="public" final="true">
    <docblock>
      <description compact="Gets the file in which the exception was created"/>
      <return type="string"/>
    </docblock>
  </method>
    <method name="getLine" abstract="false" static="false" visibility="public" final="true">
    <docblock>
      <description compact="Gets the line in which the exception was created"/>
      <return type="int"/>
    </docblock>
  </method>
    <method name="getTrace" abstract="false" static="false" visibility="public" final="true">
    <docblock>
      <description compact="Gets the stack trace"/>
      <return type="array"/>
    </docblock>
  </method>
    <method name="getTraceAsString" abstract="false" static="false" visibility="public" final="true">
    <docblock>
      <description compact="Gets the stack trace as a string"/>
      <return type="string"/>
    </docblock>
  </method>
    <method name="__toString" abstract="false" static="false" visibility="public" final="false">
    <docblock>
      <description compact="String representation of the exception"/>
      <return type="string"/>
    </docblock>
  </method>
  </parent>
  <interface full="JsonSerializable" namespace="" name="JsonSerializable">
    <method name="jsonSerialize" start="13" end="13" abstract="false" final="false" static="false" visibility="public">
      <docblock>
        <description/>
        <param type="object">
          <type full="void" namespace="" name="void"/>
        </param>
        <return type="mixed"/>
      </docblock>
    </method>
  </interface>
</class>
